auth/env_keychain.go

```diff
@@ -77,32 +98,38 @@ func ReadEnvVar(envVar string) (map[string]string, error) {
 	return authMap, nil
 }
 
-// BuildEnvVar creates the contents to use for authentication environment variable.
-//
-// Complementary to `ReadEnvVar`.
-func BuildEnvVar(keychain authn.Keychain, images ...string) (string, error) {
+func buildAuthMap(keychain authn.Keychain, images ...string) map[string]string {
 	registryAuths := map[string]string{}
 
 	for _, image := range images {
 		reference, authenticator, err := ReferenceForRepoName(keychain, image)
```

Comment:
I don't think `ReferenceForRepoName` needs to or should be exported. We could potentially inline it here?

Reply:
Sounds good. We kept the function to avoid making `buildAuthMap` too long, but we un-exported it.

Reply:
On looking into this further, it seems that `pack` is using this function. We exported it again, but we are wondering - what are the circumstances wherein we allow ourselves to make breaking changes to this package?

Reply:
Someday I expect we might have more stability at the library interface level but right now I think we can force consumers to deal with changes on upgrade as we move towards a more intentional interface that will be easier to keep stable in the long run. I am happy to leave it for now. But the logic exposed here isn't very extensive (it's basically just two ggcr function calls that would be easy to inline in pack) and isn't really specific to the lifecycle and therefore I don't see this as the kind of utility we need to expose/maintain.

All that said, happy to leave it for now in the interest of time, if you would rather.